packages/rich-list-core/src/lib/Queue.ts

```diff
@@ -0,0 +1,51 @@
+export type Mode = 'LIFO' | 'FIFO'
+
+export interface Queue<T> {
+  push: (message: T) => Promise<string> // message id / handle
+  receive: (maxMessage: number) => Promise<T[]>
+}
+
+export interface QueueClient<T> {
+  createQueueIfNotExist: (name: string, mode: Mode) => Promise<Queue<T>>
+}
+
+/**
+ * For unit test use.
+ */
```

Comment:
Since all these are for testing purposes, may be better to put them in `/test` directory for better contextual relevance?